### ISSUE TITLE
Can't install some dependencies from repository.engineering.redhat.com  when run npm install

### ISSUE THREAD

**Issue by quanzhuo** [redhat-developer/vscode-java#2874]
##### Environment
- Operating System: Ubuntu 22.04
- JDK version: 17.0.5
- Visual Studio Code version: 
- Java extension version: v1.13.0

##### Steps To Reproduce
1. clone this repo
2. checkout tag v1.13.0
3. run `npm install --verbose`

##### Current Result

Some npm packages from `https://repository.engineering.redhat.com/nexus/repository/` can't be installed due to network error. If I change these dependencies's download url to `registry.npmjs.org` in `package-lock.json`.  the `npm install` command completed OK.

##### Question

Is there some difference between packages from `repository.engineering.redhat.com` and `registry.npmjs.org` ?

**Comment by rgrunber** [comment]
Yes, in fact they should all be replaced with `registry.npmjs.org` as that's the proper approach. I wasn't able to reproduce your error though, even with a completely empty cache & node_modules. Looking into the issue, it seems like the resolved field is [only used to avoid re-fetching package metadata when it exists locally](https://github.com/npm/npm/issues/16849#issuecomment-312442508) (among other things). It also looks like your configured registry will always be used to fetch the packages (and not what's listed in the resolved field).
